sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ResolveWriteToStream.scala

```diff
@@ -81,7 +81,9 @@ object ResolveWriteToStream extends Rule[LogicalPlan] with SQLConfHelper {
           s" the query didn't fail: $tempDir. If it's required to delete it under any" +
           s" circumstances, please set ${SQLConf.FORCE_DELETE_TEMP_CHECKPOINT_LOCATION.key} to" +
           s" true. Important to know deleting temp checkpoint folder is best effort.")
-        tempDir
+        // SPARK-42676 - Write temp checkpoints for streaming queries to local filesystem
+        // even if default FS is set differently
+        "file://" + tempDir
```

Comment:
This actually broke the Windows support (https://ci.appveyor.com/project/ApacheSoftwareFoundation/spark/builds/46451196). Canonical paths are actually not necessarily a URI too - it might contain white spaces as an example.